cvat/apps/engine/filters.py

```diff
@@ -306,33 +308,38 @@ def get_lookup_fields(self, view):
 
         return get_lookup_fields(view, fields=simple_filters)
 
-    def get_schema_fields(self, view):
-        assert coreapi is not None, 'coreapi must be installed to use `get_schema_fields()`'
-        assert coreschema is not None, 'coreschema must be installed to use `get_schema_fields()`'
-
-        lookup_fields = self.get_lookup_fields(view)
+    def get_schema_operation_parameters(self, view):
+        try:
+            queryset = view.get_queryset()
+        except Exception:
+            queryset = None
+            warnings.warn(
+                "{} is not compatible with schema generation".format(view.__class__)
+            )
 
-        return [
-            coreapi.Field(
-                name=field_name,
-                location='query',
-                schema={
-                    'type': 'string',
-                }
-            ) for field_name in lookup_fields
-        ]
+        filterset_class = self.get_filterset_class(view, queryset)
 
-    def get_schema_operation_parameters(self, view):
-        lookup_fields = self.get_lookup_fields(view)
+        if not filterset_class:
+            return []
 
         parameters = []
-        for field_name in lookup_fields:
-            parameters.append({
+        for field_name, field in filterset_class.base_filters.items():
+            if isinstance(field, djf.BooleanFilter):
+                parameter_schema = { 'type': 'boolean' }
+            elif isinstance(field, (djf.NumberFilter, djf.ModelChoiceFilter)):
+                parameter_schema = { 'type': 'integer' }
+            else:
+                parameter_schema = { 'type': 'string' }
```

Comment:
Perhaps we should check that the field is actually a string here? If we're doing typing, it would be useful to check that the type we're applying is appropriate.

Reply:
Probably, these types are everything we need to support in simple filters. Basically, all fields must be serialized into strings, but we do an exception for numbers. Which other types would you like to support? To my knowledge, there are only dates that could be supported, but there are no such filters now. What is possible is that the model pk is not an integer, this is a good point.

Reply:
My point is not that we should support other types, but that when we declare a parameter to be a string, we should check that it actually is a string. If you add an assert for that, then in the future, when someone adds a filter of a type that hasn't been used before, they will be notified that they need to update the schema generation too.

Reply:
Updated